### ISSUE TITLE
getDeviceList returns devices that are no longer connected

### ISSUE THREAD

**Issue by ThijsB1983** [ksksue/FTDriver#11]
I'm running Android 4.2.2. and discovered that getDeviceList() in function begin() does not return what Android Promises ( http://developer.android.com/reference/android/hardware/usb/UsbManager.html#getDeviceList() )

Is it possible to change the begin function to check if a device 'really' is connected?

**Comment by ksksue** [comment]
Does your Android have USB Host API? Some devices do not have USB Host API even if they are upper 3.1. You might want to check it by using this app. https://play.google.com/store/apps/details?id=com.synchack.android.usbhostviewer

**Comment by ThijsB1983** [comment]
Yes, my device (a google nexus 7) has usb host API. I also added a 'openDevice' check. It's a uggly solotion that works for me. I loop trough the Devices and check if the device can be opened. Then I store the hascode of that device and us it later on in the 'begin' function.

**Comment by ksksue** [comment]
I had checked on my nexus 7 but cannot repro "getDeviceList returns devices that are no longer connected".
How to do this?

By the way, you can get broadcast receiver when USB is attached or detached
by using BroadcastReceiver (see my Android-USB-Serial-Monitor-Lite source code).

**Comment by ThijsB1983** [comment]
Strange, I plugged in and plugged out my usb device a multiple times and then the DeviceList is growing and growing....but It's more a Android bug then a bug in de FTdriver.class.

**Comment by ksksue** [comment]
Would you like to do this?
I have tested by using FTDriverTutorial1 project and Arduino UNO.
1. I have plugged in and plugged out the Arduino UNO multiple times.
2. I have tapped "Begin" button when I have plugged out it.
3. I cannot see Log.i(TAG, "Device : "+ device.toStrong()); in function begin().
It means getDeviceList returns no device that is no longer connected.
The DeviceList is not growing and growing.

**Comment by ksksue** [comment]
Does it work?

**Comment by ThijsB1983** [comment]
Sorry, i am on holyday. Will test it asap
Op 29 jul. 2013 10:09 schreef "Keisuke SUZUKI" notifications@github.com
het volgende:

**Comment by upendraaa** [comment]
Hi Ksksue, 
I have also same issue as ThijsB1983  mentioned but here I am using cp2102 cable. When I have checked same thing in sony (kitkat version) it was working fine. This problem coming with micromax funbook (4.1.2).

**Comment by ksksue** [comment]
Hi
FTDriver does not support CP2102.
Please use Physicaloid Library.

**Comment by ksksue** [comment]
Let me know which project you have checked.
